firedrake/randomfunctiongen.py

```diff
@@ -0,0 +1,251 @@
+import numpy as np
+import inspect
+import randomgen
+
+from firedrake.function import Function
+from ufl import FunctionSpace
+
+
+"""
+
+This module wraps 'randomgen' and enables
+firedrake users to pass a FunctionSpace
+to generate random functions.
+
+Usage:
+FunctionSpace, V, has to be passed as
+the first argument, e.g.:
+
+test.py:
+
+>>> from firedrake import *
+>>> mesh = UnitSquareMesh(2,2)
+>>> V = FunctionSpace(mesh, "CG", 1)
+>>> pcg = PCG64(seed=123456789)
+>>> rg = RandomGenerator(pcg)
+>>> f_beta = rg.beta(V, 1.0, 2.0)
+>>> print(f_beta.dat.data)
+[0.56462514 0.11585311 0.01247943 0.398984   0.19097059 0.5446709
+ 0.1078666  0.2178807  0.64848515]
+
+In parallel, "mpiexec -n 2 python3 test.py" shows two independent streams:
+
+[0.56462514 0.11585311 0.01247943]
+[0.03780042 0.15495691 0.42125296 0.56395561 0.07342598 0.24215732]
+
+"""
+
+
+__all__ = randomgen.__all__
+
+
+_class_names = [name for name, _ in inspect.getmembers(randomgen, inspect.isclass)]
+_method_names = [name for name, _ in inspect.getmembers(randomgen.RandomGenerator)
+                 if not name.startswith('_') and name not in ('state', 'poisson_lam_max')]
+
+
+def __getattr__(attr):
+
+    if attr == 'RandomGenerator':
+
+        class RandomGenerator(randomgen.RandomGenerator):
```

Comment:
I think we need some documentation on this too. Preferably with an autogenerated list of the RNGs available.